### PR TITLE
Add "researved" spelling correction

### PR DESCRIPTION
**Pull request by fishilico**
This mispelling occurs in several projects: https://grep.app/search?q=researved

### CODE REVIEW

File: codespell_lib/data/dictionary.txt [hunk 1]
```diff
@@ -22468,6 +22468,7 @@ rescrition->restriction
 rescritions->restrictions
 reseach->research
 reseached->researched
+researved->reserved
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:22471]: Do you want to add the reserves, reservation and reservations version of this typo too

Comment (fishilico) [reply codespell_lib/data/dictionary.txt:22473]: I added the suggested versions. Thanks

File: codespell_lib/data/dictionary.txt [hunk 2]
```diff
@@ -22468,6 +22468,10 @@ rescrition->restriction
 rescritions->restrictions
 reseach->research
 reseached->researched
+researvation->reservation
+researvations->reservations
+researved->reserved
+researves->reserves
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:22474]: Sorry @fishilico . I realised I missed the obvious one. Can you add the singular too please `researve->reserve`

Comment (fishilico) [reply codespell_lib/data/dictionary.txt:22476]: No problem. I did not know whether "researve" would be a false positive (Google shows some results for `"researve"`, including websites such as https://www.vivino.com/researve-unknown-v-7svbm/w/5990630 where it is not clear whether it is a misspelling).

On https://grep.app/, "researve" shows files that use "presearve", which seems to be a misspelling for "preserve". Therefore I added this word too.

I can squash my commits together, if you prefer.

Comment (peternewman) [reply codespell_lib/data/dictionary.txt:22476]: > No problem. I did not know whether "researve" would be a false positive (Google shows some results for `"researve"`, including websites such as https://www.vivino.com/researve-unknown-v-7svbm/w/5990630 where it is not clear whether it is a misspelling).

I'm pretty certain that's a typo too:
https://en.wikipedia.org/wiki/Reserve_wine

> On https://grep.app/, "researve" shows files that use "presearve", which seems to be a misspelling for "preserve". Therefore I added this word too.

Excellent, although you know what's coming next...

> I can squash my commits together, if you prefer.

I don't mind, indeed it's easier to review if you don't as we can see just what's changed. We can squash during the merge on GitHub if desired.

File: codespell_lib/data/dictionary.txt [hunk 3]
```diff
@@ -20069,6 +20069,7 @@ prersistent->persistent
 presance->presence
 prescrition->prescription
 prescritions->prescriptions
+presearve->preserve
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:20072]: So can we also have preserved, preservation, preserving and preserver please.

File: codespell_lib/data/dictionary.txt [hunk 4]
```diff
@@ -22468,6 +22469,11 @@ rescrition->restriction
 rescritions->restrictions
 reseach->research
 reseached->researched
+researvation->reservation
+researvations->reservations
+researve->reserve
```
Comment (peternewman) [inline codespell_lib/data/dictionary.txt:22474]: Can we have reserving too please.

File: codespell_lib/data/dictionary.txt (hunk 3 above)
Comment (fishilico) [reply codespell_lib/data/dictionary.txt:20074]: Of course :) Thanks for these ideas

**Pull request merged**